### PR TITLE
Add support for Lens Camera Profiles (LCP)

### PR DESCRIPTION
**Pull request by demoulinv**
## Description
Add lcp files support
Based on https://github.com/alicevision/AliceVision/pull/1215.

## Features list
- [X] Updates cameraInit node interface
- [X] Update ImageProcessing node interface

## Implementation remarks

- LCP info to be fullfill in cameraInit interface (database directory or filepath).
- Possibility to relax search in database by considering only the camera maker without taking into account the camera model.
- Distortion correction to be enabled in imageProcessing interface.

### CODE REVIEW

File: meshroom/nodes/aliceVision/CameraInit.py [hunk 1]
```diff
@@ -163,6 +163,27 @@ class CameraInit(desc.CommandLineNode, desc.InitNode):
...
+            label='Lens Correction Profile Info',
+            description='''Lens Correction Profile filepath or database directory.''',
+            value='${ALICEVISION_LENS_PROFILE_INFO}',
+            uid=[],
+        ),
+        desc.BoolParam(
+            name='enableLensCorrectionProfileSearch',
+            label='Enable Automatic LCP Search',
+            description='''Enable automatic LCP Search if a database directory is specified as LCP info.''',
+            value=False,
```
Comment (mugulmd) [inline meshroom/nodes/aliceVision/CameraInit.py:177]: The default value should probably be `True`

File: meshroom/nodes/aliceVision/CameraInit.py [hunk 2]
```diff
@@ -163,6 +163,27 @@ class CameraInit(desc.CommandLineNode, desc.InitNode):
...
+            uid=[0],
+        ),
+        desc.BoolParam(
+            name='lensCorrectionProfileSearchByLensNameAndCameraMakerOnly',
```
Comment (mugulmd) [inline meshroom/nodes/aliceVision/CameraInit.py:181]: This parameter name doesn't correspond to any argument of the cameraInit command-line tool, which is causing an error whenever we try to import images

Comment (mugulmd) [reply meshroom/nodes/aliceVision/CameraInit.py:181]: Same for `useInternalWhiteBalance`

**Comment by mugulmd** [comment]
Some `cameraInit` command-line arguments are missing from the `CameraInit` node parameters: 
- colorProfileDatabase
- lensCorrectionProfileSearchIgnoreCameraModel
- defaultCameraModel
- rawColorInterpretation
- errorOnMissingColorProfile
- allowSingleView

File: meshroom/nodes/aliceVision/CameraInit.py [hunk 3]
```diff
@@ -67,7 +67,7 @@
                                  " * estimated: estimated from metadata and/or sensor width. \n"
                                  " * unknown: unknown camera parameters (can still have default value guess)\n"
                                  " * none: not set",
-                     values=("calibrated", "estimated", "unknown", "none"),
+                     values=("calibrated", "estimated", "lcp", "unknown", "none"),
```
Comment (mugulmd) [inline meshroom/nodes/aliceVision/CameraInit.py:70]: New `lcp` option should also be added to the attribute description

Comment (fabiencastan) [reply meshroom/nodes/aliceVision/CameraInit.py:70]: @demoulinv ?

File: meshroom/nodes/aliceVision/CameraInit.py [hunk 4]
```diff
@@ -169,6 +169,20 @@ class CameraInit(desc.AVCommandLineNode, desc.InitNode):
...
+            name='lensCorrectionProfileInfo',
+            label='Lens Correction Profile Info',
+            description='''Lens Correction Profile filepath or database directory.''',
+            value='${ALICEVISION_LENS_PROFILE_INFO}',
+            uid=[],
+        ),
+        desc.BoolParam(
+            name='lensCorrectionProfileSearchIgnoreCameraModel',
+            label='LCP Search using lens name and camera maker only.',
+            description='''Automatic LCP Search consider only the camera model and the lens name.''',
```
Comment (fabiencastan) [inline meshroom/nodes/aliceVision/CameraInit.py:182]:
```suggestion
            description='The lens name and camera maker are used to match the LCP database, but the camera model is ignored.',
```

File: meshroom/nodes/aliceVision/CameraInit.py [hunk 5]
```diff
@@ -169,6 +169,20 @@ class CameraInit(desc.AVCommandLineNode, desc.InitNode):
...
+        desc.File(
+            name='lensCorrectionProfileInfo',
+            label='Lens Correction Profile Info',
+            description='''Lens Correction Profile filepath or database directory.''',
+            value='${ALICEVISION_LENS_PROFILE_INFO}',
+            uid=[],
+        ),
+        desc.BoolParam(
+            name='lensCorrectionProfileSearchIgnoreCameraModel',
+            label='LCP Search using lens name and camera maker only.',
```
Comment (fabiencastan) [inline meshroom/nodes/aliceVision/CameraInit.py:181]:
```suggestion
            label='LCP Generic Search',
```

File: meshroom/nodes/aliceVision/CameraInit.py [hunk 6]
```diff
@@ -67,7 +67,7 @@
                                  " * estimated: estimated from metadata and/or sensor width. \n"
```
Comment (fabiencastan) [inline meshroom/nodes/aliceVision/CameraInit.py:67]:
```suggestion
                                 " * estimated: estimated from metadata and/or sensor width \n"
                                 " * lcp: Lens Camera Profile from a database of generic calibrations \n"
```

File: meshroom/nodes/aliceVision/CameraInit.py [hunk 7]
```diff
@@ -64,7 +64,20 @@
...
+                                 " * unknown: unknown camera parameters (can still have default value guess)\n"
+                                 " * none: not set",
+                     values=("calibrated", "estimated", "unknown", "none"),
+                     value="none",
+                     exclusive=True,
+                     uid=[0],
+                     advanced=True
+                     ),
+        
+    desc.ChoiceParam(name="distoInitializationMode", label="Distortion Initialization Mode",
```
Comment (fabiencastan) [inline meshroom/nodes/aliceVision/CameraInit.py:77]: should be `distortionInitializationMode`

**Review by fabiencastan — CHANGES_REQUESTED** [review]

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**